### ISSUE TITLE
Adding XYComponent triggers assert

### ISSUE THREAD

**Issue by Normalised** [ffAudio/foleys_gui_magic#43]
When I drag an XYComponent into the View tree it always triggers an assert in juce_GraphicsContext.cpp here :

```
Rectangle<Type> coordsToRectangle (Type x, Type y, Type w, Type h) noexcept
    {
       #if JUCE_DEBUG
        const int maxVal = 0x3fffffff;

        jassert ((int) x >= -maxVal && (int) x <= maxVal
              && (int) y >= -maxVal && (int) y <= maxVal
              && (int) w >= 0 && (int) w <= maxVal
              && (int) h >= 0 && (int) h <= maxVal);
       #endif

```

To verify

1. Create a new Plugin Project in the Projucer, 
2. Add the module
3. Launch VS 2019
4. add : 

```
    AudioProcessorValueTreeState treeState{ *this, nullptr };
    foleys::MagicProcessorState magicState{ *this, treeState };
```

To PluginProcessor.h, 

5: Create the MagicPluginEditor in PluginProcessor.cpp : createEditor()

`return new foleys::MagicPluginEditor(magicState);`

6. Run the standalone in Debug 

7. drag the XYComponent from the component pallette onto the View(root) node.

This happens for me both in the master and develop branches.

**Comment by ffAudio** [comment]
Thank you for the report. Unfortunately I cannot reproduce it on my mac, I will have to restore my windows development machine, currently only the build server runs.

In the meantime you could help me a lot if you could add the call stack, so I can see, where this invalid rectangle is created.
Thanks again.

**Comment by ffAudio** [comment]
Thanks for finding that. Apparently std::atomic doesn't call the default initializer. This is fixed now with ff9c05a on the develop branch and will be included in the next release.

**Comment by ffAudio** [comment]
Fixed in release 1.1.5 today
